File: unsolved/guessinggame/guessinggame.java
import java.util.Scanner;

public class guessinggame {
	public static void main(String[] args){
		Scanner scan = new Scanner(System.in);
		int n, min, max;
		String[] line;
		while(true){
			min = 1;
			max = 10;
			n = scan.nextInt();
			if(n==0)
				break;
			scan.nextLine();
			line = scan.nextLine().split(" ");
			while(!line[1].equals("on")){
				if(line[1].equals("low"))
					min = n+1;
				else if(line[1].equals("high"))
					max = n-1;
				//System.out.println("min: "+min+"\tmax: "+max);
				n = scan.nextInt();
				scan.nextLine();
				line = scan.nextLine().split(" ");
			}
			if(n > max || n < min)
				System.out.println("Stan is dishonest");
			else
				System.out.println("Stan may be honest");
		}
		scan.close();
	}
}

File: solved/guessinggame/guessinggame.py
#!/usr/bin/python3

while True:
    n = int(input())
    if n == 0:
        break
    mini = 1
    maxi = 10
    line = input()
    while line != "right on":
        if line == "too low" and n >= mini:
            mini = n + 1
        elif line == "too high" and n <= maxi:
            maxi = n - 1
        n = int(input())
        line = input()
    if n > maxi or n < mini:
        print("Stan is dishonest")
    else:
        print("Stan may be honest")
